android-libs/src/main/java/io/ooni/mk/MKResourcesManager.java

```diff
@@ -0,0 +1,118 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+package io.ooni.mk;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.util.Log;
+
+// MKResourcesManager allows to keep resources required by C/C++ code up to date. The policy used
+// to update resources is described in the documentation of maybeUpdateResources.
+public class MKResourcesManager {
+    static final String ASN_DB = "asn.mmdb";
+    static final String CA_BUNDLE = "ca-bundle.pem";
+    static final String COUNTRY_DB = "country.mmdb";
+    static final String LOG_TAG = "MKResourcesManager";
+    static final String VERSION_JSON = "version.json";
+
+    static void copy(@NonNull InputStream source, @NonNull String dest) throws java.io.IOException {
+        org.apache.commons.io.FileUtils.copyToFile(source, new File(dest));
+    }
+
+    static boolean copySingleResource(@NonNull Context context, @NonNull String filename) {
+        String destPath = getPath(context, filename);
+        String logMsg = "Copy resource " + filename + " to " + destPath;
+        Log.v(LOG_TAG, logMsg + "...");
+        try {
+            copy(context.getAssets().open(filename), destPath);
+        } catch (java.io.IOException err) {
+            Log.e(LOG_TAG, logMsg, err);
+            return false; // something bad happened
+        }
+        Log.v(LOG_TAG, logMsg + "... done");
+        return true; // all good
+    }
+
+    static boolean copyResources(@NonNull Context context) {
+        if (!copySingleResource(context, ASN_DB)) {
+            return false; // cannot copy
+        }
+        if (!copySingleResource(context, CA_BUNDLE)) {
+            return false; // cannot copy
+        }
+        if (!copySingleResource(context, COUNTRY_DB)) {
+            return false; // cannot copy
+        }
+        if (!copySingleResource(context, VERSION_JSON)) {
+            return false; // cannot copy
+        }
+        return true; // all good
+    }
+
+    static byte[] readAll(@NonNull InputStream inputstream) throws java.io.IOException {
+        return org.apache.commons.io.IOUtils.toByteArray(inputstream);
+    }
+
+    static boolean weNeedToCopy(@NonNull Context context) {
+        try {
+            byte[] bundleVersion;
+            byte[] fsVersion;
+            // Implementation note: we assume that the version.json file is small
+            bundleVersion = readAll(context.getAssets().open(VERSION_JSON));
+            fsVersion = readAll(new FileInputStream(getVersionJSONPath(context)));
```

Comment:
So basically you copy the json as well in the filesystem and if the json in the apk is different you overwrite the resources?

Reply:
Correct